libevm/VMFactory.cpp

```diff
@@ -62,29 +75,58 @@ VMKindTableEntry vmKindsTable[] = {
     {VMKind::Hera, "hera"},
 #endif
 };
-}
 
-void validate(boost::any& v, const std::vector<std::string>& values, VMKind* /* target_type */, int)
+void setVMKind(const std::string& _name)
 {
-    // Make sure no previous assignment to 'v' was made.
-    po::validators::check_first_occurrence(v);
-
-    // Extract the first string from 'values'. If there is more than
-    // one string, it's an error, and exception will be thrown.
-    const std::string& s = po::validators::get_single_string(values);
-
     for (auto& entry : vmKindsTable)
     {
         // Try to find a match in the table of VMs.
-        if (s == entry.name)
+        if (_name == entry.name)
         {
-            v = entry.kind;
+            g_kind = entry.kind;
             return;
         }
     }
 
-    throw po::validation_error(po::validation_error::invalid_option_value);
+    // The name does not match any built-in VM, try to use it as a path to load a DLL VM.
```

Comment:
If you'd put the rest of the function into separate function, `setVMKind` itself would be simple and this comment might be not needed

Reply:
I was thinking the same so let's do it.